src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs

```diff
@@ -455,8 +460,12 @@ private static bool GetTls11Support()
             if (IsWindows)
             {
                 // TLS 1.1 and 1.2 can work on Windows7 but it is not enabled by default.
-                bool defaultProtocolSupport = !IsWindows7;
-                return GetProtocolSupportFromWindowsRegistry(SslProtocols.Tls11, defaultProtocolSupport);
+                if (IsWindows7)
+                {
+                    return GetProtocolSupportFromWindowsRegistry(SslProtocols.Tls11, false, true);
```

Comment:
Nit: can you name these bools?  It's hard to tell without that what they mean.

Reply:
updated. I also updated many comments as the logic is getting more complicated.